dask/dataframe/io/tests/test_parquet.py

```diff
@@ -2044,6 +2044,40 @@ def test_timeseries_nulls_in_schema_pyarrow(tmpdir, timestamp, numerical):
     )
 
 
+def test_read_inconsistent_schema_pyarrow(tmpdir):
+    check_pyarrow()
+
+    # Note: This is a proxy test for a cudf-related issue fix
+    # (see cudf#5062 github issue).  The cause of that issue is
+    # schema inconsistencies that do not actually correspond to
+    # different types, but whether or not the file/column contains
+    # null values.
+
+    df1 = pd.DataFrame({"id": [0, 1], "val": [10, 20]})
+    df2 = pd.DataFrame({"id": [2, 3], "val": [30, 40]})
+
+    desired_type = "int64"
+    other_type = "int32"
+    df1.val = df1.val.astype(desired_type)
+    df2.val = df2.val.astype(other_type)
+
+    df_expect = pd.concat([df1, df2], ignore_index=True)
+    df_expect['val'] = df_expect.val.astype(desired_type)
```

Comment:
Sorry, my suggestioned introduced a lint error :) Let's try this.

```suggestion
    df_expect["val"] = df_expect.val.astype(desired_type)
```